soh/soh/Enhancements/randomizer/randomizer_check_tracker.cpp

```diff
@@ -0,0 +1,1046 @@
+        startOffset += areaChecksTotal[x];
+    }
+}
+
+bool CompareCheckObject(RandomizerCheckObject i, RandomizerCheckObject j) {
+    if (i.rcArea < j.rcArea)
+        return true;
+    else if (i.rcArea > j.rcArea)
+        return false;
+
+    if (checkStatusMap.find(i.rc)->second < checkStatusMap.find(j.rc)->second)
+        return true;
+    else if (checkStatusMap.find(i.rc)->second > checkStatusMap.find(j.rc)->second)
+        return false;
+
+    if (i.rcShortName < j.rcShortName)
+        return true;
+    if (i.rcShortName > j.rcShortName)
+        return false;
+
+    return false;
+}
+
+RandomizerCheckShow GetCheckStatus(RandomizerCheckObject rcObj, int idx) {
+    if (HasItemBeenCollected(rcObj))
+        return RCSHOW_SAVED; // TODO: use SAVED until we hook into game elements without requiring a save. Then we'll use CHECKED
+
+    //If the status hasn't updated, keep showing as skipped
+    if (checkStatusMap.find(rcObj.rc)->second == RCSHOW_SKIPPED)
+        return RCSHOW_SKIPPED;
+
+    return RCSHOW_UNCHECKED;
+
+    // TODO Seen, Hinted, Scummed, saved/checked
+}
+
+
+bool HasItemBeenCollected(RandomizerCheckObject obj) {
+    ItemLocation* x = Location(obj.rc);
+    SpoilerCollectionCheck check = x->GetCollectionCheck();
+    auto flag = check.flag;
+    auto scene = check.scene;
+    auto type = check.type;
+
+    int shift;
+    int mask;
+
+    switch (type) {
+        case SpoilerCollectionCheckType::SPOILER_CHK_ALWAYS_COLLECTED:
+            return true;
+        case SpoilerCollectionCheckType::SPOILER_CHK_BIGGORON:
+            return gSaveContext.bgsFlag & flag;
+        case SpoilerCollectionCheckType::SPOILER_CHK_CHEST:
+            return gSaveContext.sceneFlags[scene].chest & (1 << flag);
+        case SpoilerCollectionCheckType::SPOILER_CHK_COLLECTABLE:
+            return gSaveContext.sceneFlags[scene].collect & (1 << flag);
+        case SpoilerCollectionCheckType::SPOILER_CHK_MERCHANT:
+        case SpoilerCollectionCheckType::SPOILER_CHK_SHOP_ITEM:
+        case SpoilerCollectionCheckType::SPOILER_CHK_COW:
+        case SpoilerCollectionCheckType::SPOILER_CHK_SCRUB:
+        case SpoilerCollectionCheckType::SPOILER_CHK_RANDOMIZER_INF:
+            return Flags_GetRandomizerInf(OTRGlobals::Instance->gRandomizer->GetRandomizerInfFromCheck(obj.rc));
+        case SpoilerCollectionCheckType::SPOILER_CHK_EVENT_CHK_INF:
+            return gSaveContext.eventChkInf[flag / 16] & (0x01 << flag % 16);
+        case SpoilerCollectionCheckType::SPOILER_CHK_GERUDO_MEMBERSHIP_CARD:
+            return CHECK_FLAG_ALL(gSaveContext.eventChkInf[0x09], 0x0F);
+        case SpoilerCollectionCheckType::SPOILER_CHK_GOLD_SKULLTULA:
+            return GET_GS_FLAGS(scene) & flag;
+        case SpoilerCollectionCheckType::SPOILER_CHK_INF_TABLE:
+            // Magic to flip an index `flag` to a lookup for 16bit big endian integers. Probably an easier way.....
+            shift = 7 - (flag % 8) + ((flag % 16) / 8) * 8;
```

Comment:
since this is used in 2 places, it'd be nice to turn this into a macro

then if someone comes up with an easier/better way it only needs to be changed in one place

Reply:
Refactored as a part of fcdf83bc2c773b5b55ae2bc79b70a2b161a238ed